polus-ftl-label-plugin/src/ftl.pyx

```diff
@@ -0,0 +1,966 @@
+            mask   = _mm_shuffle_epi8(mask,pshufbcnst)             # Align the 16x4 bit mask.
+            ind    = _mm_cvtsi128_si64(mask)                       # Extract the 16x4 bit mask.
+            ind    = ind & clear_lowest                            # Clear the lowest 4 bit address
+            count  = _mm_popcnt_u64(ind)>>2                        # Number of edges detected
+            ind    = _pext_u64(cast_index,ind)                     # Get 1-15 index offset of edges
+
+            # The next 5 intrinsics unpack the first (8) 4-bit values
+            edges     = _mm256_set1_epi64x(ind)
+            edges     = _mm256_srlv_epi64(edges,shft)
+            edges     = _mm256_and_si256(edges,vmsk)
+            edges_shf = _mm256_shuffle_epi8(edges,shf_lo)
+            edges_shf = _mm256_add_epi32(i_vec,edges_shf)
+
+            # store the results into a vector with appropriate data type for accessibility
+            _mm256_storeu_si256(<__m256i*>&temp[0],edges_shf)
+
+            # If there are more than 8 values, unpack the rest
+            if count > 8:
+                    edges_shf = _mm256_shuffle_epi8(edges,shf_hi)
+                    edges_shf = _mm256_add_epi32(i_vec,edges_shf)
+                    _mm256_storeu_si256(<__m256i*>&temp[8],edges_shf)
+
+            # store the run length encoded values
+            for i in range(count):
+                if not on_obj:
+                    on_obj=True
+                    for j in range(ndim-1):
+                        output.push_back(coords[j])
+                    output.push_back(temp[i])
+                else:
+                    on_obj=False
+                    output.push_back(temp[i])
+
+            
+            # Advance the vector index
+            i_vec      = _mm256_add_epi32(i_vec,cnst15)
+
+        # Evaluate the final n number of pixels, where n<15
+        r += 1
+        position += 1
+        for n in range(r,last_stride):
+            if image[position] == image[position-1]:
+                position += 1
+                continue
+            if on_obj:
+                on_obj = False
+                output.push_back(n)
+            else:
+                on_obj=True
+                dim = position
+                for i in range(ndim-1):
+                    output.push_back(coords[i])
+                output.push_back(n)
+            position += 1
+
+        # If the very end of the line is reached and no final edge was detected
+        if on_obj:
+            on_obj = False
+            output.push_back(last_stride)
+
+        # Reset the vector index
+        i_vec = _mm256_set1_epi16(0)
+
+    # Turn the uint16 vector into a numpy.ndarray of appropriate size
+    row_objects = np.asarray(output,dtype=np.uint32).reshape(-1,ndim+1)
+    
+    return row_objects
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.nonecheck(False)
```

Comment:
Why not have a nonecheck?

Reply:
In Cython, they include a bunch of C++ code to handle special Python cases such as None values and negative indices. Removing None checks and negative indices removes extra C++ that would get inserted every time an element in an array is access. It speeds up the code quite a bit.